clerk/tokens.go

```diff
@@ -0,0 +1,100 @@
+package clerk
+
+import (
+	"fmt"
+	"time"
+
+	"gopkg.in/square/go-jose.v2"
+
+	"gopkg.in/square/go-jose.v2/jwt"
+)
+
+type Claims struct {
+	jwt.Claims
+	UserClaims
+}
+
+type UserClaims struct {
+	Name                string `json:"name"`
+	Picture             string `json:"picture"`
+	UpdatedAt           int64  `json:"updated_at"`
+	GivenName           string `json:"given_name,omitempty"`
+	FamilyName          string `json:"family_name,omitempty"`
+	PreferredUsername   string `json:"preferred_username,omitempty"`
+	Gender              string `json:"gender,omitempty"`
+	Birthdate           string `json:"birthdate,omitempty"`
+	Email               string `json:"email,omitempty"`
+	EmailVerified       bool   `json:"email_verified,omitempty"`
+	PhoneNumber         string `json:"phone_number,omitempty"`
+	PhoneNumberVerified bool   `json:"phone_number_verified,omitempty"`
+}
+
+// DecodeToken decodes the session jwt token.
+func (c *client) DecodeToken(token string) (*Claims, error) {
+	parsedToken, err := jwt.ParseSigned(token)
+	if err != nil {
+		return nil, err
+	}
+
+	claims := Claims{}
+	if err = parsedToken.UnsafeClaimsWithoutVerification(&claims); err != nil {
+		return nil, err
+	}
+
+	return &claims, nil
+}
+
+// VerifyToken verifies the session jwt token.
+func (c *client) VerifyToken(token string) (*Claims, error) {
```

Comment:
I see the Ruby SDK accepts some additional parameters. https://github.com/clerkinc/clerk-sdk-ruby/blob/main/lib/clerk/sdk.rb#L151

Is it worth it to try something similar?

cc @agis @SokratisVidros

Reply:
Regarding the timeout, someone can set it by passing a custom http client in the sdk constructor `NewClientWithCustomHTTP()`. Although i think we should consider passing a context as the first argument to every function that has to deal with external http request. In this way we will give the developer the option to handle timeout/cancel independently for every operation. 
For the rest options and taking into account Go's way to deal with optional parameters, i don't think its worth at the current state